Bug in `OctopusDeploy_Deployment_Target`

**Describe the bug**
The resource doesn't get created and I get the below error.

**Script to reproduce**
```
resource "octopusdeploy_deployment_target" "example" {
    environments = ["Development Environment (OK to Delete)"]
    name = "Example Deployment Target"
    roles = ["Example"]
    endpoint {
        communication_style = "Kubernetes"
    }
}
```

**Expected behavior**
To make an deployment target.

**Logs and other supporting information**
```
│ Error: Plugin did not respond
│
│   with module.octopusdeploy.module.deploymenttarget_example.octopusdeploy_deployment_target.example,
│   on Modules\Infrastructure\DeploymentTargets\example.tf line 1, in resource "octopusdeploy_deployment_target" "example":
│    1: resource "octopusdeploy_deployment_target" "example" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-octopusdeploy_v0.7.58.exe plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 69 [running]:
github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy.expandKubernetesAuthentication(0x10b9d60, 0xc00033d7e0, 0x1104e1f, 0xe)
        github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy/schema_kubernetes_authentication.go:11 +0x106c
github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy.expandKubernetesCluster(0xc000670570, 0xc000670570)
        github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy/schema_kubernetes_cluster.go:13 +0x177
github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy.expandEndpoint(0xfa33c0, 0xc000666c60, 0x8, 0xfa33c0)
        github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy/schema_endpoint.go:30 +0x3d8
github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy.expandDeploymentTarget(0xc000704800, 0xc000601000)
        github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy/schema_deployment_target.go:13 +0xdd
github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy.resourceDeploymentTargetCreate(0x12299c8, 0xc000378840, 0xc000704800, 0xf856a0, 0xc00041d900, 0xc000676570, 0xb7d632, 0xc00033d020)
        github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy/resource_deployment_target.go:24 +0x52
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc000587c00, 0x1229958, 0xc0005a5680, 0xc000704800, 0xf856a0, 0xc00041d900, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/resource.go:330 +0x187
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000587c00, 0x1229958, 0xc0005a5680, 0xc000328e00, 0xc00033d020, 0xf856a0, 0xc00041d900, 0x0, 0x0, 0x0, ...)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/resource.go:456 +0x67b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00052e060, 0x1229958, 0xc0005a5680, 0xc0008ae6e0, 0xc0005a5680, 0x10b3500, 0xc0001eec00)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.1/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc00088f460, 0x1229a00, 0xc0005a5680, 0xc000328b60, 0xc00088f460, 0xc0001eec30, 0xc00053bba0)
        github.com/hashicorp/terraform-plugin-go@v0.3.1/tfprotov5/server/server.go:332 +0xb6
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x10b3500, 0xc00088f460, 0x1229a00, 0xc0001eec30, 0xc000921e00, 0x0, 0x1229a00, 0xc0001eec30, 
0xc000232a80, 0x92b)
        github.com/hashicorp/terraform-plugin-go@v0.3.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x222
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00042f6c0, 0x1232bb8, 0xc000604600, 0xc0003c5320, 0xc0002ddda0, 0x174abe0, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.40.0/server.go:1297 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00042f6c0, 0x1232bb8, 0xc000604600, 0xc0003c5320, 0x0)
        google.golang.org/grpc@v1.40.0/server.go:1626 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00009e210, 0xc00042f6c0, 0x1232bb8, 0xc000604600, 0xc0003c5320)
        google.golang.org/grpc@v1.40.0/server.go:941 +0xb2
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.40.0/server.go:939 +0x1fd

Error: The terraform-provider-octopusdeploy_v0.7.58.exe plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

--- a/octopusdeploy/schema_kubernetes_authentication.go
+++ b/octopusdeploy/schema_kubernetes_authentication.go
@@ -7,7 +7,15 @@ import (
 )
 
 func expandKubernetesAuthentication(values interface{}) octopusdeploy.IKubernetesAuthentication {
+	if values == nil {
+		return nil
+	}
+
 	flattenedValues := values.(*schema.Set)
+	if len(flattenedValues.List()) == 0 {
+		return nil
+	}
+
 	flattenedMap := flattenedValues.List()[0].(map[string]interface{})
 
 	authenticationType := flattenedMap["authentication_type"].(string)

